tests/mir-opt/simplify_pow_of_two_no_overflow_checks.rs

```diff
@@ -0,0 +1,51 @@
+// unit-test: SimplifyPowOfTwo
+// compile-flags: -Cdebug-assertions=false
+
+// EMIT_MIR simplify_pow_of_two_no_overflow_checks.slow_2_u.SimplifyPowOfTwo.after.mir
+fn slow_2_u(a: u32) -> u32 {
+    2u32.pow(a)
+}
+
+// EMIT_MIR simplify_pow_of_two_no_overflow_checks.slow_2_i.SimplifyPowOfTwo.after.mir
+fn slow_2_i(a: u32) -> i32 {
+    2i32.pow(a)
+}
+
+// EMIT_MIR simplify_pow_of_two_no_overflow_checks.slow_4_u.SimplifyPowOfTwo.after.mir
+fn slow_4_u(a: u32) -> u32 {
+    4u32.pow(a)
+}
+
+// EMIT_MIR simplify_pow_of_two_no_overflow_checks.slow_4_i.SimplifyPowOfTwo.after.mir
+fn slow_4_i(a: u32) -> i32 {
+    4i32.pow(a)
+}
+
+// EMIT_MIR simplify_pow_of_two_no_overflow_checks.slow_256_u.SimplifyPowOfTwo.after.mir
+fn slow_256_u(a: u32) -> u32 {
+    256u32.pow(a)
+}
+
+// EMIT_MIR simplify_pow_of_two_no_overflow_checks.slow_256_i.SimplifyPowOfTwo.after.mir
```

Comment:
```suggestion
// EMIT_MIR simplify_pow_of_two_no_overflow_checks.slow_256_i.SimplifyPowOfTwo.diff
```

makes it easier for us to see what the optimization does

Reply:
edit: ah, but it doesn't really optimize away anything, just replace a function... oh well 🤷

Reply:
I actually wanted to use diff as well, just for clarity's sake though it seems from stage1 to stage2 it changes an `unwind unreachable` to `unwind continue`. It's why CI is failing. Any idea why that happens/how it can be prevented? Because I'd definitely prefer diff, even just for the highlighting

Reply:
That's probably not a  stage1/stage2 difference, but a problem with the fact that we are running mir-opt tests for panic=unwind and panic=abort and are expecting the results to match. You can use `// EMIT_MIR_FOR_EACH_PANIC_STRATEGY` to generate separate output files

Reply:
Ohh, that makes sense, thanks!